vtex-apps/checkout-coupon
# CHK-172: Fix coupon input not working inside minicart

#### What problem is this solving?

This fixes an issue when this component is used within the Minicart component, where the summary wouldn't correctly pass in the `insertCoupon` and `coupon` props, although it wraps this component in the `OrderCouponProvider` from the `order-coupon` orchestrator app.

#### How to test it?

1. Access this [workspace](https://fixcoupon--checkoutio.myvtex.com/cart/add?sku=289&qty=1&seller=1).
2. Open minicart and try to apply the coupon "freeshipping".
3. The coupon should be correctly applied.

#### Describe alternatives you've considered, if any.

Another route I've considered was to fix the `checkout-summary` app props passed to this component, but since the other components from the Checkout IO project consumes these "orchestrator" apps solely via context, I prefered to change this app to be consistent with the others. Also, I chose this to avoid the prop drilling which is really fragile when working with Store Framework blocks system.

#### Related to / Depends on

N/A
